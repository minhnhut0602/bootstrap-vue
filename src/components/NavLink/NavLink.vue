<template>
    <a
        class="nav-link"
        :class="classes"
        href="#"
    >
        <slot />
    </a>
</template>

<script lang="ts" setup>
import useActive, {activeProps} from '../../composables/useActive';
import useDisabled, {disabledProps} from '../../composables/useDisabled';
import useClasses from '../../composables/useClasses';

const props = defineProps({
    ...activeProps,
    ...disabledProps,
});

const {classes} = useClasses([
    useActive(props.active).activeClass.value,
    useDisabled(props.disabled).disabledClass.value,
]);
</script>

