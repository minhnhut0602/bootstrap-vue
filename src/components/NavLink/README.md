# NavLink

```vue live
<NavLink>Default Example Usage</NavLink>
```

## Slots

| Name    | Description | Bindings |
| ------- | ----------- | -------- |
| default |             |          |
