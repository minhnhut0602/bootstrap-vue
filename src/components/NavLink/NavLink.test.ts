import NavLink from '.';

describe('template', () => {
    componentRenderTest(NavLink);

    componentSlotRenderTest(NavLink);

    componentWrapperClassTest(NavLink, {active: true}, 'active');

    componentWrapperClassTest(NavLink, {disabled: true}, 'disabled');
});
