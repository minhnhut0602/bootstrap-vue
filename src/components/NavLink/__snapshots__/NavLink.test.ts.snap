// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template renders default 1`] = `
"<a class=\\"nav-link\\" href=\\"#\\">
  <!-- @slot Displays the content of the nav link -->
</a>"
`;
